apis/mutations/v1alpha1/assignmetadata_types.go

```diff
@@ -0,0 +1,65 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// AssignMetadataSpec defines the desired state of AssignMetadata
+type AssignMetadataSpec struct {
+	Match              Match              `json:"match,omitempty"`
+	Location           string             `json:"location,omitempty"`
+	MetadataParameters MetadataParameters `json:"parameters,omitempty"`
```

Comment:
I had to name it differently to get around redeclaration of Parameters that also exist under `assign_types`. Since the Parameters structs hold different fields in each CRD type I can't use a common Parameters struct here.

Reply:
The type name has to be different, for sure.

Does the field name have to be different?

e.g.:
```
Parameters MetadataParameters `json:"parameters,omitempty"`
```